ql/src/java/org/apache/hadoop/hive/ql/io/orc/OrcSplit.java

```diff
@@ -104,13 +104,32 @@ public OrcSplit(Path path, Object fileId, long offset, long length, String[] hos
     this.isOriginal = isOriginal;
     this.hasBase = hasBase;
     this.rootDir = rootDir;
-    this.deltas.addAll(filterDeltasByBucketId(deltas, AcidUtils.parseBucketId(path)));
+    this.deltas.addAll(filterDeleteDeltasByWriteIds
```

Comment:
could we transform this construct into a stream pipeline:
````
    this.deltas = deltas.stream()
      .filter(delta -> filterDeltasByBucketId(delta, AcidUtils.parseBucketId(path)))
      .filter(delta -> filterDeleteDeltasByWriteIds(delta, conf))
      .collect(Collectors.toList());
````

Reply:
done, replaced nested function calls with stream processing.

Reply:
looks good! 
1 question, should we call addAll, or can we simply assign the result of the collect?  
````
this.deltas = ....
     collect(Collectors.toList());
````

Reply:
deltas class member is marked as final, that's why I couldn't reassign it and used addAll().